app/controllers/events_controller.rb

```diff
@@ -0,0 +1,86 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: %i[show edit update destroy]
+  before_action :require_login, except: [:index]
+
+  # GET /events or /events.json
+  def index
+    @events = Event.all
+  end
+
```

Comment:
- [ ]  Event #index should list past events, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from  [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

- [ ] Event #index should list upcoming events, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))